codegen/src/decorators/catch.rs

```diff
@@ -98,3 +98,15 @@ pub fn catch_decorator(
 
     output
 }
+
+pub fn error_decorator(
+    ecx: &mut ExtCtxt,
+    sp: Span,
+    meta_item: &MetaItem,
+    annotated: Annotatable
+) -> Vec<Annotatable> {
+    ecx.struct_span_warn(sp, "the `error` decorator has been deprecated as of Rocket v0.3.15")
```

Comment:
We should use the word "attribute" in the warning instead of decorator. I just used the word decorator in the source code because...arbitrary decision. :)

Reply:
Also, let's align this closer to the real deprecated message. Let's say:

```
warning: use of deprecated Rocket attribute `error` (deprecated since v0.3.15)
help: the `error` attribute was replaced by the `catch` attribute:`#[catch(..)]`
```

Same goes for the `errors` bang macro, of course.